### PR TITLE
(BIDS-2306) permanently harmonize proposal luck

### PR DESCRIPTION
**Pull request by LuccaBitfly**
removes luck calculation from the app (requires https://github.com/gobitfly/eth2-beaconchain-explorer/pull/2462)

### CODE REVIEW

File: src/app/tab-blocks/tab-blocks.page.ts [hunk 1]
```diff
@@ -146,13 +146,10 @@ export class TabBlocksPage implements OnInit {
...
 				`Compares the number of your actual proposed blocks to the expected average blocks per validator during the last <strong>${
-					this.luck.timeFrameName
+					this.luck.timeFrameName ? this.luck.timeFrameName : 'month'
 				}</strong>. 
-        <br/><br/>Your <strong>${
-					this.luck.userValidators
-				}</strong> validators are expected to produce <strong>${this.luck.expectedBlocksPerMonth.toFixed(
-					2
-				)}</strong> blocks per month on average with current network conditions.`
+        <br/><br/>Your ${this.luck.userValidators == 1 ? `validator is` : `<strong>${this.luck.userValidators}</strong> validators are`}
```
Comment (D13ce) [inline src/app/tab-blocks/tab-blocks.page.ts:151]: +1 :+1:

**Review by D13ce — CHANGES_REQUESTED** [review]
Really nice changes :+1:

I have left some comments though and at least one of them requires a change (removing a now deprecated function). ptal

**Review by D13ce — CHANGES_REQUESTED** [review]
As mentioned here https://github.com/gobitfly/eth2-beaconchain-explorer/pull/2462#pullrequestreview-1582340429, we will have to refactor this a bit:

The new endpoint only returning the luck will be removed and the luck will be returned as part of the dashboard endpoint.

**Review by jonesk7734 — APPROVED** [review]
😃

File: src/app/utils/BlockUtils.ts [hunk 2]
```diff
@@ -81,147 +72,27 @@ export class BlockUtils extends CacheModule {
...
-		if (timeframe == -1) return null
-
-		const blocksAfterTimeframe = this.getBlocksAfterTs(blocks, timeframe)
-		if (blocksAfterTimeframe.length <= 0) return null
-
-		const avgBlockInTimeframe = this.calculateExpectedBlocksInTimeframe(timeframe, currentEpoch.validatorscount, valis.length)
+		if (valis.length == 0) return null
 
+		const request = new ProposalLuckRequest(...valis.map((vali) => vali.index))
+		const response = await this.api.execute(request)
```
[the remark this answers is not in the document]
Comment (D13ce) [reply src/app/utils/BlockUtils.ts:80]: The luck is now also returned via the dashboard API endpoint, see https://github.com/gobitfly/eth2-beaconchain-explorer/pull/2462

File: src/app/utils/BlockUtils.ts [hunk 3]
```diff
@@ -62,16 +53,16 @@ export class BlockUtils extends CacheModule {
 	}
 
 	async getMyBlocks(offset: number): Promise<BlockResponse[]> {
-		const valis = await this.validatorUtils.getLocalValidatorIndexes()
+		const valis = await this.validatorUtils.getAllValidatorsLocal()
```
[the remark this answers is not in the document]
Comment (LuccaBitfly) [reply src/app/utils/BlockUtils.ts:56]: removed

File: src/app/utils/BlockUtils.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (LuccaBitfly) [reply src/app/utils/BlockUtils.ts:80]: This is not a problem anymore I guess?

**Review by D13ce — APPROVED** [review]
lgtm

**Pull request merged**